sdl2-sonic-drivers/src/files/dmx/OP2File.cpp

```diff
@@ -0,0 +1,70 @@
+#include <files/dmx/OP2File.hpp>
+#include <cstring>
+
+namespace files
+{
+    namespace dmx
+    {
+        constexpr const char* OP2FILE_MAGIC_HEADER = "#OPL_II#";
+        constexpr int OP2FILE_MAGIC_HEADER_SIZE = 8;
+        constexpr int OP2FILE_INSTRUMENT_NAME_MAX_SIZE = 32;
+        constexpr int OP2FILE_INSTRUMENT_NUM_VOICES = 2;
+
+        OP2File::OP2File(const std::string& filename) : File(filename)
+        {
+            // header
+            char buf[OP2FILE_MAGIC_HEADER_SIZE + 1];
+            read(buf, OP2FILE_MAGIC_HEADER_SIZE);
+            buf[OP2FILE_MAGIC_HEADER_SIZE] = 0;
+            _assertValid(strncmp(buf, OP2FILE_MAGIC_HEADER, OP2FILE_MAGIC_HEADER_SIZE) == 0);
+
+            // instruments
+            _readInstruments();
+
+            // instrument names
+            _readInstrumentNames();
+        }
+
+        OP2File::instrument_t OP2File::getInstrument(const int i) const noexcept
+        {
+            return _instruments[i];
```

Comment:
need to check the i value is not out of range.

replace with ``.at(i)` and remove `noexcept`

Reply:
also replace int with uint8_t